modules/aws-velero/iam.tf

```diff
@@ -54,3 +57,33 @@ resource "aws_iam_policy" "velero_backup" {
 }
 EOF
 }
+
+resource "aws_iam_role" "velero_backup" {
+  count = length(var.oidc_provider_url) != 0 ? 1 : 0
+  name = "${var.name}-${var.env}-velero-backup"
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/${var.oidc_provider_url}"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "${var.oidc_provider_url}:sub": "system:serviceaccount:kube-system:velero"
+                }
+            }
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "velero_backup" {
+  count = length(var.oidc_provider_url) != 0 ? 1 : 0
```

Comment:
Some brainstorming. Can we extract length(var.oidc_provider_url) != 0 ? 1 : 0 to a named local value?